Smartling.API/Job/JobApiClient.cs

```diff
@@ -78,35 +78,37 @@ public virtual AddFileToJobResponse AddLocale(string localeId, string jobId, boo
       return JsonConvert.DeserializeObject<AddFileToJobResponse>(response["response"]["data"].ToString());
     }
 
-    public virtual List<Smartling.Api.Model.Job> Get(string jobName = "")
+    public virtual List<Smartling.Api.Model.Job> Get(string jobName = "", IEnumerable<string> allowedStatuses = null)
     {
-      var page = GetPage(jobName, PageSize, 0);
+      var page = GetPage(jobName, PageSize, 0, allowedStatuses);
       var results = new List<Smartling.Api.Model.Job>();
       var pageNumber = 0;
       results.AddRange(page.items);
 
       while (page.totalCount > results.Count && page.items.Count > 0)
       {
         pageNumber++;
-        page = GetPage(jobName, PageSize, PageSize * pageNumber);
+        page = GetPage(jobName, PageSize, PageSize * pageNumber, allowedStatuses);
         results.AddRange(page.items);
       }
 
       return results;
     }
-
-    public virtual JobList GetPage(string jobName, int limit, int offset)
+    
+    public virtual JobList GetPage(string jobName, int limit, int offset, IEnumerable<string> allowedStatuses = null)
     {
-      StringBuilder uriBuilder;
-      if (string.IsNullOrEmpty(jobName))
+      var url = string.Format(GetJobUrl, projectId, limit, offset);
+      if (!string.IsNullOrEmpty(jobName))
       {
-        uriBuilder = this.GetRequestStringBuilder(string.Format(GetJobUrl, projectId, limit, offset));
+        url += $"&jobName={System.Net.WebUtility.UrlEncode(jobName)}";
       }
-      else
+
+      if (allowedStatuses != null)
       {
-        uriBuilder = this.GetRequestStringBuilder(string.Format(GetJobByNameUrl, projectId, jobName, limit, offset));
+        url = allowedStatuses.Aggregate(url, (current, allowedStatus) => current + $"&translationJobStatus={System.Net.WebUtility.UrlEncode(allowedStatus)}");
       }
 
+      var uriBuilder = this.GetRequestStringBuilder(url);
```

Comment:
It looks a little bit strange to create string builder on the last step. It seems you can easily use uriBuilder in two conditional statements above.

Reply:
Updated.